src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/Cache.java

```diff
@@ -136,34 +136,61 @@ public void put(
         CacheEntry oldCacheEntry = cachedEntries.get(requestUrlHash);
         if (oldCacheEntry == null)
         {
-            CacheEntry cacheEntry = new CacheEntry(
+            final CacheEntry cachedEntry;
+
+            if (request.protocolStackHeader())
+            {
+                cachedEntry = new EmulatedCacheEntry(
+                    this,
+                    request,
+                    true,
+                    supplyTrace);
+            }
+            else
+            {
+                cachedEntry = new NativeCacheEntry(
                     this,
                     request,
                     true,
                     supplyTrace);
-            updateCache(requestUrlHash, cacheEntry);
+            }
+
+
+            updateCache(requestUrlHash, cachedEntry);
         }
         else
         {
             boolean expectSubscribers = (request.getType() == Type.INITIAL_REQUEST) || oldCacheEntry.expectSubscribers();
-            CacheEntry cacheEntry = new CacheEntry(
+
+            final CacheEntry cachedEntry;
+            if (request.protocolStackHeader())
+            {
+                cachedEntry = new EmulatedCacheEntry(
                     this,
                     request,
                     expectSubscribers,
                     supplyTrace);
-
-            if (cacheEntry.isIntendedForSingleUser())
+            }
+            else
+            {
+                cachedEntry = new NativeCacheEntry(
+                    this,
+                    request,
+                    expectSubscribers,
+                    supplyTrace);
+            }
```

Comment:
Since this if statement logic is coming up more than once, let's add a `newCacheEntry(...)` method to return either an `EmulatedCacheEntry` or a `DefaultCacheEntry`. (was `NativeCacheEntry`)